tests/python/gpu/test_fusion.py

```diff
@@ -337,3 +338,36 @@ def test_fusion_reshape_executor():
     out = f.forward(is_train=False, data1=data, data2=data)
     assert out[0].sum().asscalar() == 150
 
+@with_seed()
+def test_input_reorder():
+    class Block(gluon.HybridBlock):
+        def __init__(self, **kwargs):
+            super(Block, self).__init__(**kwargs)
+
+        def hybrid_forward(self, F, x, y, z):
+            s = x * 2
```

Comment:
Would would happen without this fix? Does gluon try to feed inputs in incorrect order and sometimes give wrong result?

Reply:
Without this fix fusion would generate a graph with different inputs order which would be caught and the fusion would be disabled.